readthedocs/restapi/views/task_views.py

```diff
@@ -29,10 +29,12 @@ def get_status_data(task_name, state, data, error=None):
         'data': data,
         'started': state in STARTED_STATES,
         'finished': state in FINISHED_STATES,
-        'success': state in SUCCESS_STATES,
+        # When an exception is raised inside the task, we keep this as SUCCESS
+        # and add the exception messsage into the 'error' key
+        'success': state in SUCCESS_STATES and error is None,
     }
-    if error is not None and isinstance(error, Exception):
-        data['error'] = error.message
+    if error is not None:
+        data['error'] = error
```

Comment:
Similarly, we should probably we defensive here about what error is. Are we sure it will always be a string type, or should we be actively turning it into a string here, also?

Reply:
It should be always a string since that `data` object here is the `info` dictionary that we are setting in the `run` method of the task where we cast the Exception object to a string.